### PR TITLE
Create problems.rb

### PR DESCRIPTION
**Pull request by hmc-cs-ctan**
@AdamSchulze  help

### CODE REVIEW

File: problems.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+def confusing_function_that_has_bugs(times)
```
Comment (cvalleroy) [inline problems.rb]: Needs a more readable title

Comment (AdamSchulze) [inline problems.rb]: This function name could be a bit shorter, and it would help if it said a little bit about what the function does.

Comment (cvalleroy) [inline problems.rb]: Needs a more clear title

File: problems.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  # TODO: Fix this whole function
+  total_times = 4 * n
+  for i in 0...n
+    for j in 0...3
+      puts “Hi”
```
Comment (AdamSchulze) [inline problems.rb]: Storing "Hi" in a variable would allow us to avoid magic values.

Comment (cvalleroy) [inline problems.rb]: You could store "Hi" in a variable to be called

File: problems.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+def confusing_function_that_has_bugs(times)
+  # TODO: Fix this whole function
+  total_times = 4 * n
+  for i in 0...n
+    for j in 0...3
```
Comment (AdamSchulze) [inline problems.rb]: Combining these two loops into one loop would improve clarity

File: problems.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+def confusing_function_that_has_bugs(times)
+  # TODO: Fix this whole function
```
Comment (AdamSchulze) [inline problems.rb]: TODO comments should be removed.

File: problems.rb (hunk 3 above)
Comment (cvalleroy) [inline problems.rb]: Consider merging the nested for loops into one loop, improves clarity

File: problems.rb (hunk 4 above)
Comment (cvalleroy) [inline problems.rb]: Need to remove TODO comments before pull

**Review by AdamSchulze — CHANGES_REQUESTED** [review]
See comments below. Some changes are necessary.

File: problems.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+  total_times = 4 * n
```
Comment (cvalleroy) [inline problems.rb]: You haven't used total_times at all yet!

**Review by AdamSchulze — APPROVED** [review]

**Pull request merged**